Core/Core/Quizzes/QuizDetails/View/QuizEditorView.swift

```diff
@@ -0,0 +1,335 @@
+                    value: $viewModel.allowedAttempts
+                )
+            }
+        }
+    }
+
+    @ViewBuilder
+    private var oneQuestionSection: some View {
+        EditorSection {
+            ToggleRow(
+                label: Text("Show One Question at a Time", bundle: .core),
+                value: $viewModel.oneQuestionAtaTime)
+            if viewModel.oneQuestionAtaTime {
+                Divider()
+                ToggleRow(
+                    label: Text("Lock Questions After Answering", bundle: .core),
+                    value: $viewModel.lockQuestionAfterViewing)
+            }
+        }
+    }
+
+    @ViewBuilder
+    private var accessCodeSection: some View {
+        EditorSection {
+            ToggleRow(
+                label: Text("Require an Access Code", bundle: .core),
+                value: $viewModel.requireAccessCode)
+            if viewModel.requireAccessCode {
+                Divider()
+                TextFieldRow(
+                    label: Text("Access Code", bundle: .core),
+                    placeholder: NSLocalizedString("Enter code", comment: ""),
+                    text: $viewModel.accessCode
+                )
+            }
+        }
+    }
+
+    @ViewBuilder
+    private var assignmentOverridesSection: some View {
+        AssignmentOverridesEditor(
+            courseID: viewModel.courseID,
+            groupCategoryID: viewModel.assignment?.groupCategoryID,
+            overrides: $viewModel.assignmentOverrides,
+            toRemove: Binding(get: {
+                if case .removeOverride(let override) = alert {
+                    return override
+                }
+                return nil
+            }, set: {
+                alert = $0.map { AlertItem.removeOverride($0) }
+            })
+        )
+    }
+
+    @ViewBuilder
+    private var quizTypeRow: some View {
+        ButtonRow(action: {
+            let options = viewModel.availableQuizTypes
+            self.env.router.show(ItemPickerViewController.create(
+                title: NSLocalizedString("Quiz Type", comment: ""),
+                sections: [ ItemPickerSection(items: options.map {
+                    ItemPickerItem(title: $0.name)
+                }), ],
+                selected: options.firstIndex(of: viewModel.quizType).flatMap {
+                    IndexPath(row: $0, section: 0)
+                },
+                didSelect: { viewModel.quizType = options[$0.row] }
+            ), from: controller)
+        }, content: {
+            Text("Quiz Type", bundle: .core)
+            Spacer()
+            Text(viewModel.quizType.name)
+                .font(.medium16).foregroundColor(.textDark)
+            Spacer().frame(width: 16)
+            DisclosureIndicator()
+        })
+    }
+
+    @ViewBuilder
+    private func assignmentGroupRow(assignmentGroup: AssignmentGroup) -> some View {
+        ButtonRow(action: {
```

Comment:
This closure is a good candidate to be rather part of the viewmodel as a function.

Reply:
Great point, I'll move these!